diff --git a/CARL/CMakeLists.txt b/CARL/CMakeLists.txt
index 2319dc9..262fab9 100644
--- a/CARL/CMakeLists.txt
+++ b/CARL/CMakeLists.txt
@@ -33,6 +33,6 @@ set_property(TARGET carl_core PROPERTY FOLDER "CARL")
 if (MSVC)
     target_compile_options(carl_core PRIVATE /bigobj)
 endif()
-if(ANDROID)
+if(ANDROID OR UNIX)
     set_property(TARGET carl_core PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
diff --git a/Targets/CMakeLists.txt b/Targets/CMakeLists.txt
index 764ffca..764196d 100644
--- a/Targets/CMakeLists.txt
+++ b/Targets/CMakeLists.txt
@@ -13,6 +13,8 @@ if(EMSCRIPTEN)
     target_link_options(carl_static_library PUBLIC -s EXPORTED_FUNCTIONS=["_getBytes","_startRecording","_recordInputSample","_finishRecording","_serializeRecording","_deserializeRecording","_disposeRecording","_getRecordingInspector","_inspect","_getStartTimestamp","_getEndTimestamp","_disposeRecordingInspector","_createdDefinition","_addExample","_getDefaultSensitivity","_setDefaultSensitivity","_serializeDefinition","_deserializeDefinition","_getExamplesCount","_getRecordingFromExampleAtIdx","_getStartTimestampFromExampleAtIdx","_getEndTimestampFromExampleAtIdx","_disposeDefinition","_createSession","_addInputSample","_disposeSession","_createRecognizer","_getCurrentScore","_setSensitivity","_getCanonicalRecordingInspector","_disposeRecognizer","_malloc","_free"])
 elseif(WIN32 OR WINDOWS_STORE)
     target_compile_definitions(carl_static_library PRIVATE CARL_PLATFORM_WINDOWS)
+elseif(UNIX)
+    target_compile_definitions(carl_static_library PRIVATE CARL_PLATFORM_LINUX)
 elseif(ANDROID)
     target_compile_definitions(carl_static_library PRIVATE CARL_PLATFORM_ANDROID)
     target_link_libraries(carl_static_library PRIVATE ${android_logging})
@@ -25,6 +27,9 @@ if(NOT EMSCRIPTEN)
     set_property(TARGET carl_shared_library PROPERTY FOLDER "Targets")
     if(WIN32 OR WINDOWS_STORE)
         target_compile_definitions(carl_shared_library PRIVATE CARL_PLATFORM_WINDOWS)
+    elseif(UNIX)
+        target_compile_definitions(carl_shared_library PRIVATE CARL_PLATFORM_LINUX)
+        set_property(TARGET carl_shared_library PROPERTY POSITION_INDEPENDENT_CODE ON)
     elseif(ANDROID)
         target_compile_definitions(carl_shared_library PRIVATE CARL_PLATFORM_ANDROID)
         target_link_libraries(carl_shared_library PRIVATE ${android_logging})
diff --git a/Targets/c_api/main.cpp b/Targets/c_api/main.cpp
index a05e9b5..bbf6bea 100644
--- a/Targets/c_api/main.cpp
+++ b/Targets/c_api/main.cpp
@@ -13,6 +13,12 @@
 #include <android/log.h>
 #endif
 
+#ifdef __GNUC__
+#ifndef __cdecl
+#define __cdecl
+#endif
+#endif
+
 #ifdef CARL_PLATFORM_WINDOWS
 #define C_API_EXPORT(ReturnT) __declspec(dllexport) ReturnT __cdecl
 #define C_API_CALLBACK(ReturnT) ReturnT __cdecl
diff --git a/Targets/console/main.cpp b/Targets/console/main.cpp
index baef8f8..dca671c 100644
--- a/Targets/console/main.cpp
+++ b/Targets/console/main.cpp
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#include <carl/carl.h>
+#include <carl/Carl.h>
 
 #include <filesystem>
 #include <fstream>
